### ISSUE TITLE
How to handle file uploads?

### ISSUE THREAD

**Issue by thought2** [panter/ra-data-prisma#23]
The generated CRUD (by `addCrudResolvers`) assumes that there's a mapping from the input to the output types. But with a file upload, this pattern is broken as far as I can see. The input type will be some `Upload` scalar and the output type some custom `File` type. 

Or do I miss something here?

I was trying to implement some file upload following [this example](https://github.com/gaspan/graphql-prisma2-file-upload/).

**Comment by macrozone** [comment]
its indeed something that is not handled properly at the moment. Also prisma does not support files directly in the db.

We used two workarounds:

- storing file as base64 string. The client would already upload the base64 string. For displaying the image, i would recommend to create server route that can return the image, instead of passing the base64 string back to clients. Storing files as base64 in the db increases the space required on the database and performance is not that good for large files. But depending on your requirements, it can be good enough. It has the advantage that you have everything in the database

- uploading the files to S3 (AWS or similar) BEFORE updating a record in react-admin. You would create a cutom Fileinput that uploads the file you add to it directly to S3 and has the file url as string value. This is also very simple and performant but might lead to orphaned files on the S3 (but as s3 is cheap, we simply ignored that...)
